mypy/main.py

```diff
@@ -174,6 +174,9 @@ def process_options(args: List[str],
     parser.add_argument('--strict-optional', action='store_true',
                         dest='special-opts:strict_optional',
                         help="enable experimental strict Optional checks")
+    parser.add_argument('--strict-optional-whitelist', metavar='GLOB', nargs='*',
+                        help="suppress strict Optional errors in all but the provided files"
+                        " (experimental)")
```

Comment:
Make this stronger, e.g. `" (experimental -- read documentation before using!)"`